src/algorithm/hnswlib/hnswalg.h

```diff
@@ -1329,7 +1329,7 @@ class HierarchicalNSW : public AlgorithmInterface<float> {
                 link_lists_[i] = nullptr;
             } else {
                 element_levels_[i] = link_list_size / size_links_per_element_;
-                link_lists_[i] = (char*)malloc(link_list_size);
+                link_lists_[i] = (char*)allocator_->Allocate(link_list_size);
```

Comment:
Do you review and replace all malloc(..) allocations?

Reply:
I had added unit tests to make sure that all the allocations and deallocations is matching.